### PR TITLE
Enabled flake8 to run on tests

### LINKED ISSUE #106: Make tests pass flake8 and alter pipelines.
I would like to start enforcing that tests pass flake8.

### PR DESCRIPTION
**Pull request by davfsa**
- Fix flake8 issues on tests

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
Closes #106

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -15,18 +15,20 @@ ignore =
...
     S101,      # Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
 
-# F401:  unused import.
-# F403:  cannot detect unused vars if we use starred import
+# F401: unused import.
+# F403: cannot detect unused vars if we use starred import
 # FS003: f-string missing prefix.
 # R102: unnecessary parenthesis on raised exception (raises false positives in places)
+# S106: posible hardcoded password (we dont care about this in the tests)
+# D104, D103, D101, D100: Missing docstring in public function/class/module (ignore in tests)
```
Comment (nekokatt) [inline .flake8:23]: We should add docstrings.

File: .flake8 [hunk 2]
```diff
@@ -15,18 +15,20 @@ ignore =
...
+# PT011: Set the match parameter for `pytest.raises`
```
Comment (nekokatt) [inline .flake8:24]: What is this?

**Comment by nekokatt** [comment]
Can't approve on mobile for some reason but LGTM.

**Review by nekokatt — APPROVED** [review]

File: .flake8 (hunk 2 above)
Comment (davfsa) [reply .flake8:24]: There is a parameter you can pass to `pytest.warn` that allows you to make sure the error that was raised is the expected error. We don't use it so i just ignored it. If you want I can try to implement this

File: .flake8 (hunk 1 above)
Comment (davfsa) [reply .flake8:23]: They are too much hassle for what they bring. The test name should describe extacly what the test tests. If any further info is needed, there are comments explaining it

File: .flake8 (hunk 2 above)
Comment (nekokatt) [reply .flake8:24]: go for it

File: .flake8 (hunk 1 above)
Comment (nekokatt) [reply .flake8:23]: Hmm. It would enable some of the tests to be clearer and have shorter names IMO.

Comment (davfsa) [reply .flake8:23]: It's going to be more of a PITA cause I would have to write a docstring for each of the 1000 tests...

Comment (nekokatt) [reply .flake8:23]: :-)

Comment (nekokatt) [reply .flake8:23]: We should still consider doing this at some point.

**Comment by nekokatt** [comment]
Please rebase onto master :)

I have marked this as draft until you do that, as you said there was something else you"d like to do when you get time on this PR.

**Comment by davfsa** [comment]
@nekokatt Did what I wanted to do :+1:

**Review by nekokatt — APPROVED** [review]

**Pull request merged**